postpic/_field_calc.py

```diff
@@ -242,3 +242,90 @@ def divE(self, **kwargs):
         ret.name = 'div E'
         ret.shortname = 'divE'
         return ret
+
+    def kspace(self, component, omega_func=None):
+        '''
+        Reconstruct the physical kspace of one polarization component
+
+        self must be a self object containing the data.
+        The required fields will be read from the self.
+
+        component must be one of ["Ex", "Ey", "Ez", "Bx", "By", "Bz"].
+
+        omega_func may be a function that will calculate the dispersion
+        relation of the simulation. The function will receive one argument
+        that contains the k mesh.
+
+        This function basically computes one component of
+            E = 0.5*(E - omega/k^2 * Cross[k, E])
+        or
+            B = 0.5*(B + 1/omega * Cross[k, B])
+        after removing the grid stagger.
+        '''
+        # target field is polfield and the other field is otherfield
+        polfield = component[0]
+        otherfield = 'B' if polfield == 'E' else 'E'
+
+        # polarization axis
+        polaxis = helper.axesidentify[component[1]]
+
+        # build a dict of all the Field-getter functions of the dump reader
+        field_factories = {'E': {0: self.Ex, 1: self.Ey, 2: self.Ez},
+                           'B': {0: self.Bx, 1: self.By, 2: self.Bz}}
+
+        # get all the needed fields...
+        fields = []
+
+        # the polaxis component of polfield
+        fields.append(field_factories[polfield][polaxis]())
+
+        # and the other two components of the otherfield needed for the cross-product
+        fields.append(field_factories[otherfield][(polaxis+1) % 3]())
+        fields.append(field_factories[otherfield][(polaxis+2) % 3]())
+
+        # find the origin of the simulation box, needed to unstagger the fields
+        dims = self.simdimensions()
+        simorigin = [self.simextent(i)[0] for i in range(dims)]
```

Comment:
E and B fields may cover only a part of the simulation box. Therefore the simorigin is quite meaningless. Probably better is to use the extent of the first field given.

Reply:
Changed to rely on first given field